cmd/web-handlers.go

```diff
@@ -728,12 +728,25 @@ func (web *webAPIHandlers) ListAllBucketPolicies(r *http.Request, args *ListAllB
 	if !isHTTPRequestValid(r) {
 		return toJSONError(errAuthentication)
 	}
+	// FIXME: remove this code when S3 layer for gateway and server is unified.
+	var policyInfo policy.BucketAccessPolicy
+	var err error
 
-	policyInfo, err := readBucketAccessPolicy(objectAPI, args.BucketName)
-	if err != nil {
-		return toJSONError(err, args.BucketName)
+	switch layer := objectAPI.(type) {
+	case *s3Objects:
```

Comment:
Convert this to a function and use it in both SetBucketPolicy and ListAllBucketPolicies.